src/server/Completions/AttributeCompletionFactory.ts

```diff
@@ -44,33 +51,63 @@ export default class AureliaAttributeCompletionFactory extends BaseAttributeComp
    * Look at the View Model and provide all class variables as completion in kebab case.
-   * TODO: Only bindables should be provided.
    */
   private addViewModelBindables(result: CompletionItem[], elementName: string, aureliaApplication: AureliaApplication) {
-    if (aureliaApplication.components.length > 0) {
-      aureliaApplication.components.forEach(component => {
-        if (component.name !== elementName || component.viewModel?.properties === undefined) return;
-
-        component.viewModel.properties.forEach(property => {
-          if (!property.isBindable) return;
-
-          const documentation = `Bindable type: \`${property.type}\``;
-
-          const quote = this.settings.quote;
-          const varAsKebabCase = property.name.split(/(?=[A-Z])/).map(s => s.toLowerCase()).join('-');
-          result.push({
-            documentation: {
-              kind: MarkupKind.Markdown,
-              value: documentation
-            },
-            detail: 'Au View Model Bindable',
-            insertText: `${varAsKebabCase}.$\{1:bind}=${quote}$\{0:${property.name}}${quote}`,
-            insertTextFormat: InsertTextFormat.Snippet,
-            kind: CompletionItemKind.Variable,
-            label: `Au View Model: ${varAsKebabCase}`
+    const program = aureliaApplication.getProgram();
+    if (program !== undefined) {
+      const checker = program.getTypeChecker();
+      for (const sourceFile of program.getSourceFiles()) {
+        const elementNameCandidate = Path.basename(sourceFile.fileName).replace(/\.(ts|js|html)$/, '').split(/(?=[A-Z])/).map(s => s.toLowerCase()).join('-');
+        if (!sourceFile.isDeclarationFile && elementNameCandidate === elementName) {
+          sourceFile.forEachChild((node) => {
+            if (ts.isClassDeclaration(node) && node.name.getText().split(/(?=[A-Z])/).map(s => s.toLowerCase()).join('-') === elementName) {
+              node.members.forEach(member => {
+                if (ts.isPropertyDeclaration(member) && this.propertyIsBindable(member)) {
+                  const propertyName = member.name?.getText();
+                  const bindableType = `Bindable type: \`${member.type?.getText() !== undefined ? member.type?.getText() : "unknown" }\``;
+
+                  // Add comment documentation if available
+                  const symbol = checker.getSymbolAtLocation(member.name);
+                  const commentDoc = ts.displayPartsToString(
+                    symbol.getDocumentationComment(checker)
+                  );
+
+                  // Add default values
+                  const defaultValue = `Default value: \`${member.initializer?.getText()}\``;
+
+                  // Concatenate documentation parts with spacing
+                  const documentation = `${commentDoc}\n\n${bindableType}\n\n${defaultValue}`;
+
+                  const quote = this.settings.quote;
+                  const varAsKebabCase = propertyName.split(/(?=[A-Z])/).map(s => s.toLowerCase()).join('-');
```

Comment:
I have kebabCase somewhere lying around, might make sense to replace this one now
(I know you just copy pasted)

Reply:
Yup, it's a copy paste from the old solution. I fear regex with a passion, so if it worked I thought I could just reuse it. We could have a `ElementUtil` or the equivalent where we have methods such as `getElementNameFromClassDeclaration` and `getElementNameFromSourceFile`?

Reply:
Yup definitely, at the start, getting things to work makes sense